File: leetcode/easy/27-RemoveElement.py

def removeElement(nums, val):
    k = 0
    for i in range(0, len(nums)):
        if nums[i] != val:
            nums[k] = nums[i]
            k += 1
    return k

nums = [3,2,2,3]
val = 2
res = removeElement(nums, val)

print(res)
for i in range(0, res):
    print(nums[i], end=" ")

File: leetcode/easy/27-RemoveElement.cpp
#include<bits/stdc++.h>
using namespace std;


int removeElement(vector<int>& nums, int val) {
    int k = 0;
    for (int i=0; i<nums.size(); i++) {
        if (nums[i] != val) {
            nums[k++] = nums[i];
        }
    }

    return k;
}

int main() {
    vector<int>nums = {3,2,2,3};
    int val = 3;

    int res = removeElement(nums, val);

    cout << res << endl;
    for(int i=0; i<res; i++)
        cout << nums[i] << " ";

    return 0;
}
